packages/widget/src/controllers/transfers/substrate/build.ts

```diff
@@ -0,0 +1,40 @@
+import { SubstrateAssetTransfer } from '@buildwithsygma/sygma-sdk-core/substrate';
+import type { ApiPromise } from '@polkadot/api';
+import { type FungibleTokenTransferController } from '../fungible-token-transfer';
+
+export async function buildSubstrateFungibleTransactions(
+  this: FungibleTokenTransferController
+): Promise<void> {
+  const address = this.walletContext.value?.substrateWallet?.signerAddress;
+  if (
+    !this.sourceNetwork ||
+    !this.destinationNetwork ||
+    !this.resourceAmount ||
+    !this.selectedResource ||
+    !this.destinatonAddress ||
+    !address
+  ) {
+    return;
+  }
+
+  const substrateTransfer = new SubstrateAssetTransfer();
+  await substrateTransfer.init(
+    this.substrateProviderContext.value?.substrateProvider as ApiPromise,
```

Comment:
Aren't you ignoring user selected source network here?

Reply:
if I understand you correctly, we should init api promise based on the user selection, which makes total sense. Perhaps have the user to provide the collection of connections of multiple providers, user selects source and then upon matching them, we initialize the api promise object. Wdty?